libs/sql-parser/src/main/java/io/crate/sql/Literals.java

```diff
@@ -124,8 +124,8 @@ public static String replaceEscapedChars(String input) {
                         i = endIdx - 1; // skip already consumed chars
                         break;
                     default:
-                        // non-valid escaped char sequence
-                        builder.append(currentChar);
+                        // Apply rule of the C-Style escaped sequence: any other character following a backslash is taken literally.
+                        // Current char is backslash, we ignore it. Only nextChar must be added, it will be done in the next iteration.
```

Comment:
Then, this is a bit confusing, we don't ignore the backslash if next char is not a backslash? right?
so if `\\%` we ignore the first backslash right?

Reply:
Next character doesn't matter here (when switch came to default) as it means all cases where next char matters are not hit. So here we always ignore current backslash, regardless what's next.

For double backslash we have [dedicated check](https://github.com/crate/crate/blob/master/libs/sql-parser/src/main/java/io/crate/sql/Literals.java#L77)

For  sequence of slashes, basically rule of thumb is 
show `floor(slash_count/2)` slashes and then add next char. 

Some tests of CrateDB on this PR and PG 15 (both return same results)
```
 select E'\%' as one; ==> '%;
 select E'\\%' as two; ==> '\%';
 select E'\\\%' as three; ==> '\%';
 select E'\\\\%' as four; ==> '\\%';
 select E'\\\\\%' as five; ==> '\\%';
 select E'\\\\\\%' as six; ==> '\\\%';
```

Reply:
Thx, I was confused with the context of the switch, all good!